src/utils/OpenProXLLF.cpp

```diff
@@ -750,109 +815,87 @@ class TOPPOpenProXLLF :
 
-//########################### TESTING SCORES ##############################################
+        // These fields are not written yet, so at lest avoid random values by initializing to 0
+        csm.PScoreCommon = 0;
+        csm.PScoreXlink = 0;
+        csm.PScoreBoth = 0;
+        csm.PScoreAlpha = 0;
+        csm.PScoreBeta = 0;
 
-//        map<Size, PeakSpectrum> peak_level_spectra = PScore::calculatePeakLevelSpectra(spectrum, rankMap[scan_index]);
-//        csm.PScoreCommon = PScore::computePScore(fragment_mass_tolerance, fragment_mass_tolerance_unit_ppm, peak_level_spectra, theoretical_spec_common);
-//        csm.PScoreXlink = PScore::computePScore(fragment_mass_tolerance, fragment_mass_tolerance_unit_ppm, peak_level_spectra, theoretical_spec_xlinks);
-//        csm.PScoreBoth = PScore::computePScore(fragment_mass_tolerance, fragment_mass_tolerance_unit_ppm, peak_level_spectra, theoretical_spec);
-//        csm.PScoreAlpha = PScore::computePScore(fragment_mass_tolerance, fragment_mass_tolerance_unit_ppm, peak_level_spectra, theoretical_spec_alpha);
-//        csm.PScoreBeta = PScore::computePScore(fragment_mass_tolerance, fragment_mass_tolerance_unit_ppm, peak_level_spectra, theoretical_spec_beta);
-
-//        csm.HyperCommon = HyperScore::compute(fragment_mass_tolerance, fragment_mass_tolerance_unit_ppm, spectrum, theoretical_spec_common);
-//        csm.HyperAlpha = HyperScore::compute(fragment_mass_tolerance, fragment_mass_tolerance_unit_ppm, spectrum, theoretical_spec_alpha);
-//        csm.HyperBeta = HyperScore::compute(fragment_mass_tolerance, fragment_mass_tolerance_unit_ppm, spectrum, theoretical_spec_beta);
-//        csm.HyperXlink = HyperScore::compute(fragment_mass_tolerance_xlinks, fragment_mass_tolerance_unit_ppm, spectrum, theoretical_spec_xlinks);
-//        csm.HyperBoth = HyperScore::compute(fragment_mass_tolerance, fragment_mass_tolerance_unit_ppm, spectrum, theoretical_spec);
-
-//########################### END TESTING SCORES ###########################################
-
-
-#ifdef _OPENMP
-#pragma omp critical (max_subscore_variable_access)
-#endif
-        {
-          if (xcorrx_max > xcorrxMax) xcorrxMax = xcorrx_max;
-          if (xcorrc_max > xcorrcMax) xcorrcMax = xcorrc_max;
-        }
+        csm.HyperCommon = 0;
+        csm.HyperAlpha = 0;
+        csm.HyperBeta = 0;
+        csm.HyperXlink = 0;
+        csm.HyperBoth = 0;
 
         // Compute score from the 4 scores and 4 weights
+        // The weights are adapted from the xQuest algorithm (O. Rinner et al., 2008, "Identification of cross-linked peptides from large sequence databases"),
+        // they were determined by an Linear Discriminant Analysis on CID fragmentation data.
+        // The match-odds score does not work very well on HCD data and label-free cross-linkers (has the maximal possible value very often), so its weight was drastically reduced here.
```

Comment:
For this version of the tool we wanted a reimplementation of the original algorithm.
We are currently working on an improved scoring scheme, but that will take a while.
Also all the subscores are included in the output, so if anyone wants to do their own weighting, they can do that already.